### DISCUSSION TITLE
Any way to only have one UAC prompt when updating packages?

### DISCUSSION THREAD

**Discussion by demonbane** [Devolutions/UniGetUI#594] [Q&A]
Is there any way to "cache" the UAC check for WingetUI itself so that I don't have to click approve for each package being updated? It looks like gsudo has an option for this, but the regular "Update selected packages" workflow doesn't appear to call gsudo.

**Comment by marticliment** [comment]
This is a think that should be discussed seriously.

I think it could be potentially dangerous to chain run installers as administrator.

You can run wingetui as admin to achieve this result, but it is not recommended at all.

This feature could lead people to run software with admin privileges without noticing, which, in my opinion, is a security flaw. What do you think?

**Comment by ppvnf** [comment]
I think this is related to https://github.com/marticliment/WingetUI/issues/520

**Comment by demonbane** [comment]
I agree that letting WingetUI do this automatically would be a bad idea, but I think doing a single UAC check per **user action** (like clicking "Update selected packages" for example) would be a good compromise. As soon as that action completes, any further actions would prompt another UAC check.

**Reply by marticliment** [reply]
I agree on that. This could be the solution

**Comment by Stephan-P** [comment]
And maybe turn this into a setting that each user can enable/disable
